### PR TITLE
Update `cl` list commands

### PR DESCRIPTION
**Pull request by maxwang7**
Uses do_command to call do_help_command
Issue: #686

### CODE REVIEW

**Comment by kashizui** [comment]
This ends up breaking functionality for `cl <command> -h` or `cl <command> --help`, which should show just the help for a single command. What about just updating line 180 to call `do_command(...)`?

**Review by kashizui — CHANGES_REQUESTED** [review]
This ends up breaking functionality for `cl <command> -h` or `cl <command> --help`, which should show just the help for a single command. What about just updating line 180 to call `do_command(...)`?

**Comment by maxwang7** [comment]
I added an extra check so that `cl --help` and `cl -h` would both produce the help information as well. It feels a bit hacky but I couldn't find anywhere else that seemed more appropriate to add this code.

time: 40 minutes (wow, a lot longer than expected...)

File: codalab/lib/bundle_cli.py [hunk 1]
```diff
@@ -176,8 +176,8 @@ def print_help(self, out_file=None):
     def error(self, message):
         # Adapted from original:
         # https://hg.python.org/cpython/file/2.7/Lib/argparse.py
-        if len(sys.argv) == 1:
-            self.print_help()
```
Comment (percyliang) [inline codalab/lib/bundle_cli.py:180]: Can you delete `print_help` now?

**Review by percyliang — APPROVED** [review]
Make sure all the variants of --help, help, -h with and without other commands work.

**Comment by maxwang7** [comment]
Confirmed that all variants work. I tried these:

- `cl`
- `cl -h`
- `cl --help`
- `cl help`
- `cl add --help`
- `cl add -h`

**Review by kashizui — APPROVED** [review]

**Pull request merged**